src/common/opencl.c

```diff
@@ -624,20 +623,41 @@ static gboolean _opencl_device_init(dt_opencl_t *cl,
     {
       dt_print_nts(DT_DEBUG_OPENCL,
                    "  *** could not get platform vendor for device `%s' : %s\n",
-                   cl->dev[dev].name, cl_errstr(err));
+                   device_name, cl_errstr(err));
       g_strlcpy(platform_vendor, "???", DT_OPENCL_CBUFFSIZE);
     }
   }
 
+  if(has_platform_name)
+    g_strlcpy(platform_display_name, platform_name, DT_OPENCL_CBUFFSIZE);
+  else
+    g_strlcpy(platform_name, "unknownplatform", DT_OPENCL_CBUFFSIZE);
+
+  // get the canonical platform name and device name
+  platform_cname_size = strlen(platform_name) + 1;
```

Comment:
@jenshannoschwalm While the device_name is retrieved using `dt_opencl_get_device_info` that gets and checks for valid `param_value_size`,  the code to get the platform_name doesn't populate the last  `clGetPlatformInfo` param_value_size. Since the opencl api says that the returned string is null terminated I just used `strlen` here.

Is it ok for you or is there a reason to do something similar as done in `dt_opencl_get_device_info`?

Reply:
Seems ok for me too.